src/MSBuild.Abstractions/MSBuildHelpers.cs

```diff
@@ -185,6 +185,26 @@ public static bool IsWinForms(IProjectRootElement projectRoot)
             return DesktopFacts.KnownWinFormsReferences.Any(reference => references.Contains(reference, StringComparer.OrdinalIgnoreCase));
         }
 
+        /// <summary>
+        /// Determines if a given project is a WinUI Project by looking at PackageReferences
+        /// TODO: Change this method? is theis the best way to check If it is WinUI?
+        /// </summary>
+        /// <param name="projectRoot"></param>
+        /// <returns></returns>
+        public static bool IsWinUI(IProjectRootElement projectRoot)
+        {
+            var packageReferences = projectRoot.ItemGroups.SelectMany(GetPackageReferences)?.Select(elem => elem.Include.Split(',').FirstOrDefault());
```

Comment:
i think the best way to tell if it is a **UWP** project is to look at the `TargetPlatformIdentifier` property. If it is set to `UAP` then you know it is a UWP (I know, the difference between UAP and UWP is confusing)

Reply:
Updated method